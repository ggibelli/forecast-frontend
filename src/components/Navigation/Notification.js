import React from 'react'
import { useSelector } from 'react-redux'
import { makeStyles } from '@material-ui/core/styles'
import Alert from '@material-ui/lab/Alert'

const Notification = () => {
  const notification = useSelector((state) => state.notification)
  const useStyles = makeStyles((theme) => ({
    root: {
      width: '100%',
      '& > * + *': {
        marginTop: theme.spacing(2),
      },
    },
  }))
  const classes = useStyles()

  if (!notification) {
    return null
  }

  return (
    <div className={classes.root}>
      <Alert severity={notification.type}>{notification.message}</Alert>
    </div>
  )
}

export default Notification
