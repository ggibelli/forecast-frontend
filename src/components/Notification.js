import React from 'react'
import { useSelector } from 'react-redux'

const Notification = () => {
  const notification = useSelector((state) => state.notification)

  if (!notification) {
    return null
  }

  const style = {
    borderStyle: 'solid',
    borderRadius: 5,
    padding: 10,
    color: notification.type === 'success' ? 'green' : 'red',
    background: 'lightgrey',
  }

  return <div style={style}>{notification.message}</div>
}

export default Notification
