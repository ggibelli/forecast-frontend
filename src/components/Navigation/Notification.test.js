import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer'

import Notification from './Notification'

const mockStore = configureMockStore([thunk])
let store
let component

describe('Notification', () => {
  beforeEach(() => {
    store = mockStore({
      notification: { message: 'ciao', type: 'success' },
    })
    component = renderer.create(
      <Provider store={store}>
        <Notification />
      </Provider>,
    )
  })
  test('renders Notification component', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
