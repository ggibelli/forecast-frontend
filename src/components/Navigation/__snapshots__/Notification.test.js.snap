// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders Notification component 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
    role="alert"
  >
    <div
      className="MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message"
    >
      ciao
    </div>
  </div>
</div>
`;
